integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonPreAggregateRules.scala

```diff
@@ -98,238 +143,499 @@ case class CarbonPreAggregateQueryRules(sparkSession: SparkSession) extends Rule
-               metaData.hasAggregateDataMapSchema =>
-          val (carbonTable, tableName) = getCarbonTableAndTableName(logicalRelation)
-          // if it is valid plan then extract the query columns
-          isValidPlan = extractQueryColumnsFromAggExpression(groupingExp,
-            aggregateExp,
-            carbonTable,
-            tableName,
-            list)
-          carbonTable
+  /**
+   * Below method will be used to update the child plan
+   * This will be used for updating expression like join condition,
+   * order by, project list etc
+   * @param plan
+   * child plan
+   * @return updated plan
+   */
+  def updatePlan(plan: LogicalPlan) : LogicalPlan = {
+    val updatedPlan = plan transform {
+      case Aggregate(grp, aggExp, child) =>
+        Aggregate(updateExpression(grp), updateNamedExpression(aggExp), child)
+      case Filter(filterExp, child) =>
+        Filter(updateConditionExpression(Some(filterExp)).get, child)
+      case Project(projectList, child) =>
+        Project(updateNamedExpression(projectList), child)
+      case Sort(sortOrders, global, child) =>
+        Sort(updateSortExpression(sortOrders), global, child)
+      case Join(left, right, joinType, condition) =>
+        Join(left, right, joinType, updateConditionExpression(condition))
+    }
+    updatedPlan
+  }
 
-        // below case for handling filter query
-        // When plan has grouping expression, aggregate expression
-        // filter expression
-        case Aggregate(groupingExp,
-          aggregateExp,
-          Filter(filterExp,
-          CarbonSubqueryAlias(_, logicalRelation: LogicalRelation)))
-          // only carbon query plan is supported checking whether logical relation is
-          // is for carbon
-          if logicalRelation.relation.isInstanceOf[CarbonDatasourceHadoopRelation]   &&
-             logicalRelation.relation.asInstanceOf[CarbonDatasourceHadoopRelation].carbonRelation.
-               metaData.hasAggregateDataMapSchema =>
-          val (carbonTable, tableName) = getCarbonTableAndTableName(logicalRelation)
-          // if it is valid plan then extract the query columns
-          isValidPlan = extractQueryColumnsFromAggExpression(groupingExp,
-            aggregateExp,
-            carbonTable,
-            tableName,
-            list)
-          if(isValidPlan) {
-            isValidPlan = !CarbonReflectionUtils.hasPredicateSubquery(filterExp)
-          }
-          // getting the columns from filter expression
-          if(isValidPlan) {
-            isValidPlan = extractQueryColumnFromFilterExp(filterExp, list, carbonTable, tableName)
+  /**
+   * Below method will be used to update the sort expression
+   * @param sortExp
+   * sort order expression in query
+   * @return updated sort expression
+   */
+  def updateSortExpression(sortExp : Seq[SortOrder]) : Seq[SortOrder] = {
+     sortExp map { order =>
+      order.child match {
```

Comment:
Better do the transform instead of match as in case if Alias comes match cannot work